### DISCUSSION TITLE
splitting datas in 101 databases

### DISCUSSION THREAD

**Discussion by deepakchandarr** [stuttter/ludicrousdb#131] [Q&A]
I would like to shard data in between 101 databases, one global db, and 100 blog DB for WordPress multisite . its like the last 2 digits of the blog id should decide which database it needs to be settled,
 for example,
    We can predict what database a blog is in by looking 
 * at the last two digits of the id. Examples:
 *
 * + blog_id: 9, in db09
 * + blog_id: 74, in db74
 * + blog_id: 999989, in db89
 * + blog_id: 9200, in db00

and i used this call back , 
function pb_db_callback( $query, $wpdb ) {
  if ( preg_match( "/^{$wpdb->base_prefix}\d+_/i", $wpdb->table ) ) {
    $last_two_digits = (int) substr( $wpdb->blogid, -2 );
    $db = sprintf( 'db%02d', $last_two_digits ); // db00, db01, db02, ..., db99
    return $db;
  } else {
    return 'global';
  }
}
$wpdb->add_callback( 'pb_db_callback' );

but im getting error , can anyone help me out in setting this up. 
i took reference from a blog whcih use ludicrous db " [here](https://pressbooks.org/blog/2017/10/19/moving-half-a-million-database-tables-to-aws-aurora-part-1/) "

**Comment by takotakot** [comment]
> getting error

Please explain the errors that you got.

**Reply by deepakchandarr** [reply]
im getting a critical error (but didn't get any mail about the error) when db.php in installed in the site

**Reply by deepakchandarr** [reply]
and i also think , there might be a mistake from my side for configuration but cant find where,i installed this as mentioned then simply added a global database in main database, then rest 100 databases in the blog database and used that callback function, I didn't do anything else .please let me know if i need to configure anything else ?
wp version : 5.7.3
no of tables per subsite(multisite) = 60

**Reply by takotakot** [reply]
Please try

1. ludicrousdb with one database
2. ludicrousdb with primary - secondary config

before complicated settings.

And please show all `db-config.php` later.

**Reply by deepakchandarr** [reply]
hi , i setit up again , but this time it didnt show error, but it acts like nothing has changed, everything stores in default  wpdb which was configured in wp-config.php, and its not storing in the secondary databases
here is my configuration


$wpdb->add_database( array(
	'host'     => 'localhost',
	'user'     => 'ownsite_mur',
	'password' => 'password',
	'name'     => 'ownsite_mdb',
) );

$wpdb->add_database( array(
	'host'     => 'localhost',
	'user'     => 'ownsite_dbuser',
	'password' => 'password',
	'name'     => 'ownsite_db01',
	'dataset'  => 'blog',
) );

$wpdb->add_callback( 'my_db_callback' );

// Multisite blog tables are "{$base_prefix}{$blog_id}_*"
function my_db_callback( $query, $wpdb ) {
	if ( preg_match("/^{$wpdb->base_prefix}\d+_/i", $wpdb->table) ) {
		return 'blog';
	}
}

**Reply by deepakchandarr** [reply]
here is exactly what i did,
1. copied "ludicrousdb" folder in plugin dir "wp-content/plugins/ludicrousdb/"
2. again copied these files to the respective directories, "wp-content/db.php" , "wp-content/db-error.php" & "ABSPATH/db-config.php"(db-config.php on the same directory as the wp-config located)
3. then opened "ABSPATH/db-config.php" and added the database and callback functions back as mentioned above message .
4. i tried the above callback with 2 database , global and secondary but its not woring,
5. then i tried with 101 databases, one global and 100 blog databases and used the call back function as mentioned in my very first message . but same result.

its like after adding db_config also the wp is using the old configuration.
other than this i didnt change anything

**Reply by deepakchandarr** [reply]
> Please try
> 
> 1. ludicrousdb with one database
> 2. ludicrousdb with primary - secondary config
> 
> before complicated settings.
> 
> And please show all `db-config.php` later.

hi  . this didnt work ,please help me out in this

**Comment by deepakchandarr** [comment]
it works now ,
